### ISSUE TITLE
config's `filter` parser does not support multiline strings

### ISSUE THREAD

**Issue by TriplEight** [nextest-rs/nextest#868]
[TOML](https://toml.io/en/v1.0.0#string) s[upports](https://learnxinyminutes.com/docs/toml/) both multiline and multiline literal strings.

Whereas trying with a simple `\` or just a new line in a literal sting I get 
```
error: failed to parse nextest config at `scripts/nextest.toml`

Caused by:
  newline in string found at line 130 column 157 in scripts/nextest.toml
```

And with both multiline and multiline literal strings I get
```
error: expected end of expression
   ,-[1:1]
 1 | ,-> test(benchmark_block_works) | test (adder::stress_spawn) | test(adder::execute_good_chain_on_parent) | test(adder::execute_good_block_on_parent) |
 2 | |-> test(adder::execute_bad_block_on_parent) | test(it:execute_queue_doesnt_stall_with_varying_executor_params) | test(it:deleting_prepared_artifact_does_not_dispute)
   : `---- unparsed input
   `----
```

Rationale: I've got a lot of wickedly-named tests that require special treatment and putting them in one line would be a horror.

Also, it takes quite some trial and error to make the parser happy. Thankfully, `show-config test-groups` gives the output like
```
note: to see what the problems were, use the option `--future-incompat-report`, or run `cargo report future-incompatibilities --id 5`
group: flaky (max threads = 1)
  * override for default profile with filter 'test(benchmark_block_works) | test (adder::stress_spawn) | test(adder::execute_good_chain_on_parent) | test(adder::execute_good_block_on_parent) | test(adder::execute_bad_block_on_parent) | test(it:execute_queue_doesnt_stall_with_varying_executor_params) | test(it:deleting_prepared_artifact_does_not_dispute)':
      benchmark_block:
          benchmark_block_works
      node-core-pvf-worker::it:
          adder::execute_bad_block_on_parent
          adder::execute_good_block_on_parent
          adder::execute_good_chain_on_parent
          adder::stress_spawn
```
But it would be cool to have a warning or error (better if the user could control it) that would tell that not the entire `filter` line was parsed. Or something is not found and why.

**Comment by sunshowers** [comment]
Yeah, we should support multiline strings. Should be an easy fix -- probably will get to it some time in the next week.

**Comment by sunshowers** [comment]
re your second concern, you can see which tests match a given filter by using `cargo nextest list -E '<filter>'`.

**Comment by TriplEight** [comment]
> re your second concern, you can see which tests match a given filter by using `cargo nextest list -E '<filter>'`.

yeah, but it requires to build the project to get the tests list anyway. So I guess without this list it won't be possible to get a match right? Otherwise it could be just a mere static check that the expression is valid, which you already have.
Maybe spit this list out in ~ `nextest run -v` when it meets a `filter` or `-E`?

**Comment by sunshowers** [comment]
Implemented in #875, planning to release this tomorrow.

**Comment by sunshowers** [comment]
This will be out as cargo-nextest 0.9.53 within the next 30 minutes.
